peak/family.py

```diff
@@ -51,6 +51,8 @@ def assemble(self, locals, globals): pass
     @abstractmethod
     def get_adt_t(self, adt_t): pass
 
+    @abstractmethod
```

Comment:
Do we need both get_adt_t and get_constructor?

Reply:
Unfortunately we do since from_fields is a different constructor from __init__ . Perhaps in the future we can expand __init__ to support the same arguments as from_fields

Reply:
No I understand that but is their a time when we would want init and not from fields?

Reply:
Well there are times I (developer) want the AADT type (so I use get_adt_t), but I guess I could construct it manually. I dont care.

Reply:
if you want its fine, just wanted to make sure if there was a usecase.  We might want to make it an internal method though or something to not confuse users.